src/Evolution/Systems/GeneralizedHarmonic/Initialize.hpp

```diff
@@ -0,0 +1,195 @@
+      GeneralizedHarmonic::Tags::ConstraintGamma0Compute<Dim, Inertial>,
+      GeneralizedHarmonic::Tags::ConstraintGamma1Compute<Dim, Inertial>,
+      GeneralizedHarmonic::Tags::ConstraintGamma2Compute<Dim, Inertial>>;
+
+  template <typename DbTagsList, typename... InboxTags, typename Metavariables,
+            typename ArrayIndex, typename ActionList,
+            typename ParallelComponent>
+  static auto apply(db::DataBox<DbTagsList>& box,
+                    const tuples::TaggedTuple<InboxTags...>& /*inboxes*/,
+                    const Parallel::ConstGlobalCache<Metavariables>& /*cache*/,
+                    const ArrayIndex& /*array_index*/,
+                    const ActionList /*meta*/,
+                    const ParallelComponent* const /*meta*/) noexcept {
+    // First add 3+1 quantities to the box
+    auto _box =
+        db::create_from<db::RemoveTags<>, db::AddSimpleTags<>, compute_tags>(
+            std::move(box));
+
+    // Then compute gauge related quantities
+    const size_t num_grid_points =
+        db::get<::Tags::Mesh<Dim>>(_box).number_of_grid_points();
+
+    // fetch lapse, shift, spatial metric and their derivs through compute tags
+    const auto& lapse = get<gr::Tags::Lapse<DataVector>>(_box);
+    const auto& dt_lapse = get<::Tags::dt<gr::Tags::Lapse<DataVector>>>(_box);
+    const auto& deriv_lapse =
+        get<::Tags::deriv<gr::Tags::Lapse<DataVector>, tmpl::size_t<Dim>,
+                          Inertial>>(_box);
+    const auto& shift = get<gr::Tags::Shift<Dim, Inertial, DataVector>>(_box);
+    const auto& dt_shift =
+        get<::Tags::dt<gr::Tags::Shift<Dim, Inertial, DataVector>>>(_box);
+    const auto& deriv_shift =
+        get<::Tags::deriv<gr::Tags::Shift<Dim, Inertial, DataVector>,
+                          tmpl::size_t<Dim>, Inertial>>(_box);
+    const auto& spatial_metric =
+        get<gr::Tags::SpatialMetric<Dim, Inertial, DataVector>>(_box);
+    const auto& trace_extrinsic_curvature =
+        get<gr::Tags::TraceExtrinsicCurvature<DataVector>>(_box);
+    const auto& trace_christoffel_last_indices = get<
+        gr::Tags::TraceSpatialChristoffelFirstKind<Dim, Inertial, DataVector>>(
+        _box);
+
+    // call compute item for the gauge source function
+    using ExtraVars = db::item_type<extras_tag>;
+    ExtraVars extra_vars{num_grid_points};
+    get<GeneralizedHarmonic::Tags::GaugeH<Dim, Inertial>>(extra_vars) =
+        GeneralizedHarmonic::Tags::GaugeHImplicitFrom3p1QuantitiesCompute<
+            Dim, Inertial>::function(lapse, dt_lapse, deriv_lapse, shift,
+                                     dt_shift, deriv_shift, spatial_metric,
+                                     trace_extrinsic_curvature,
+                                     trace_christoffel_last_indices);
+
+    // set time derivatives of GaugeH = 0
+    const auto& dt_gauge_source =
+        make_with_value<tnsr::a<DataVector, Dim, Inertial>>(lapse, 0.);
```

Comment:
For consistency the time derivative should be chosen according to the gauge that is used. Of course the result should be the same here, but this being a prototype for future GHG evolutions, I think we should implement the general case.

Reply:
- the original comment here by @hrueter from 18days ago was for the line above in which `dt_gauge_source` was being explicitly initialized to `0`.
 - Now `dt_gauge_source = dt<GaugeH>` is provided through the compute tag `GeneralizedHarmonic::SpacetimeDerivDampedHarmonicHCompute`. @geoffrey4444 - `SpacetimeDerivGaugeHCompute` is NO longer used here.
 - Now it is `dt<Initial data's GaugeH>`, which is needed by the damped harmonic gauge, that is being initialized to `0`. github has simply moved over this conversation to the line initializing `dt_initial_gauge_source`.

Reply:
In real life, I think we'll only have two cases: i) analytic solutions, where we will always want the initial gauge to be stationary, and ii) numerical solutions, where we will read in the time derivatives. For now, I favor hard-coding zero here with a comment noting that this will have to be extended to handle numerical initial data or analytic initial data whose gauge is not initially stationary.